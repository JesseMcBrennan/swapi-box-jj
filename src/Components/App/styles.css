.App {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 1em;
}

.loading-gif {
  margin: 1em;
  border: 2px solid rgba(255, 211, 0);
  border-radius: .5em;
  opacity: .8;
}