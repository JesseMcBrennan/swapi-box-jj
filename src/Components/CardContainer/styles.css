.card-container {
  display: flex;
  overflow: auto;
  width: 100%;
}