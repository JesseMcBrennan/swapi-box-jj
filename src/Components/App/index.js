
import React, { Component } from 'react';
import './styles.css';
import { filmDataCall } from '../../data/apiFetchCall';
import { IntroContainer } from '../IntroContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
      filmData: []
    };
  }

  componentDidMount() {
    this.retrieveFilmData();
  }

  retrieveFilmData = async () => {
    const filmData = await filmDataCall();
    this.setState({
      filmData
    });
  }

  render() {
    const { filmData } = this.state;
    const displayData = filmData.length ? <IntroContainer filmData={filmData} /> : <h1>place holder</h1>;
    return (
      <div className="App">
        {displayData}
      </div>
    );
  }
}

export default App;
