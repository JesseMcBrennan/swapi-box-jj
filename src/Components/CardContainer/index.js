import React from 'react';
import PropTypes from 'prop-types';
import { Card } from "../Card";
import './styles.css';

export const CardContainer = ({ peopleData, planetData, vehicleData, favoritesData }) => {
  // const displayPeopleCards = peopleData.map(person => <Card {...person} />);

  return (
    <div>
   {/*   {displayPeopleCards}*/}
    </div>
  );
};

CardContainer.propTypes = {
  peopleData: PropTypes.array
};