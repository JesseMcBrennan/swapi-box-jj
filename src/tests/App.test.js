import React from 'react';
import { shallow } from 'enzyme';
import App from '../Components/App';
import CardContainer from '../Components/CardContainer';
import { mockMovieData } from '../data/__mocks__/api-call';
import { mockCleanedFilmData } from '../data/__mocks__/data-cleaner';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders the App with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call retrieveFilmData when componentDidMount is invoked', async () => {
    // const spy = jest.spyOn(wrapper.instance(), 'retrieveFilmData');
    // await wrapper.instance().componentDidMount();
    // expect(spy).toHaveBeenCalled();
  });

  it('should update state with filmData when retrieveFilmData is called', async () => {
    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        ok: true,
        json: () => Promise.resolve(mockMovieData)
      }));
    await wrapper.instance().retrieveFilmData();
    expect(wrapper.state('filmData')).toEqual(mockCleanedFilmData);
  });

  describe('displaySelection tests', () => {
    it('should return a CardContainer with peopleData', () => {
      // const mockPeopleData = [{key1 : 'value'}, {key2 : 'value'}];
      // const mockToggleFavorites = jest.fn();
      // const mockCardsToDisplay = 'people';
      // const result = wrapper.instance().displaySelection(mockCardsToDisplay);
      // const expected = shallow(
      //   <CardContainer 
      //     cardsData={mockPeopleData} 
      //     toggleFavorites={mockToggleFavorites}
      //   />);
      // expect(wrapper.find(CardContainer).prop('cardsData')).toEqual(wrapper.instance().displaySelection);
    });
  });
});
