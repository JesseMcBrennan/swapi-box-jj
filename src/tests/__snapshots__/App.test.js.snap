// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the App with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="header"
>
          SWAPI-Box
</h1>,
        <Unknown
          favoritesLength={0}
          handleClick={[Function]}
          selectedButton=""
/>,
        <div>
          <img
                    src="loading.gif"
          />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWAPI-Box",
          "className": "header",
        },
        "ref": null,
        "rendered": "SWAPI-Box",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "favoritesLength": 0,
          "handleClick": [Function],
          "selectedButton": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            src="loading.gif"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "loading.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="header"
>
            SWAPI-Box
</h1>,
          <Unknown
            favoritesLength={0}
            handleClick={[Function]}
            selectedButton=""
/>,
          <div>
            <img
                        src="loading.gif"
            />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAPI-Box",
            "className": "header",
          },
          "ref": null,
          "rendered": "SWAPI-Box",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "favoritesLength": 0,
            "handleClick": [Function],
            "selectedButton": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="loading.gif"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "loading.gif",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
